music/src/core/data.ts

```diff
@@ -799,3 +800,170 @@ export class MultitrackConverter extends DataConverter {
     return noteSequence;
   }
 }
+
+/**
+ * Converts to and from hit/velocity/offset representations.
+ * In this setting, we represent drum sequences and performances
+ * as triples of (hit, velocity, offset). Each timestep refers to a fixed beat
+ * on a grid, which is by default spaced at 16th notes (when `stepsPerQuarter`
+ * is 4). Drum hits that don't fall exactly on beat are represented through the
+ * offset value, which refers to the relative distance from the nearest
+ * quantized step.
+ *
+ * Hits are binary [0, 1].
+ * Velocities are continuous values in [0, 1].
+ * Offsets are continuous values in [-0.5, 0.5], rescaled to [-1, 1] for
+ * tensors.
+ *
+ * @param stepsPerQuarter The number of quantization steps per quarter note.
+ * @param humanize If True, flatten all input velocities and microtiming. The
+ * model then learns to map from a flattened input to the original sequence.
+ * @param fixedVelocities: If True, flatten all input velocities.
+ */
+export interface GrooveConverterArgs extends BaseConverterArgs {
+  stepsPerQuarter?: number;
+  humanize?: boolean;
+  tapify?: boolean;
+  pitchClasses?: number[][];
+  splitInstruments?: boolean;
+}
+export class GrooveConverter extends DataConverter {
+  readonly stepsPerQuarter: number;
+  readonly humanize: boolean;
+  readonly pitchClasses: number[][];
+  readonly pitchToClass: Map<number, number>;
+  readonly depth: number;
+  readonly endTensor: tf.Tensor1D;
+  readonly splitInstruments: boolean;
+
+  constructor(args: GrooveConverterArgs) {
+    super(args);
+
+    this.stepsPerQuarter =
+        args.stepsPerQuarter || constants.DEFAULT_STEPS_PER_QUARTER;
+    this.pitchClasses = args.pitchClasses || DEFAULT_DRUM_PITCH_CLASSES;
+    this.pitchToClass = new Map<number, number>();
+    for (let c = 0; c < this.pitchClasses.length; ++c) {  // class
+      this.pitchClasses[c].forEach((p) => {
+        this.pitchToClass.set(p, c);
+      });
+    }
+    this.humanize = args.humanize;
+    this.splitInstruments = args.splitInstruments;
+
+    // Each drum hit is represented by 3 numbers -
+    // on/off, velocity, and offset
+    this.depth = 3;
```

Comment:
What does depth mean though? Or, why is it called depth?

Reply:
This is variable is described in the abstract class.